spec/components/navbar/navbar_component_spec.rb

```diff
@@ -0,0 +1,38 @@
+require "rails_helper"
+
+describe Navbar::NavbarComponent, type: "component" do
+  let(:all_frontmatter) do
+    { "/mock-path" => { title: "Mock title", navigation: 20 } }
+  end
+
+  before do
+    allow(Pages::Frontmatter).to receive(:list).and_return all_frontmatter
+  end
+
+  subject! { render_inline(Navbar::NavbarComponent.new(all_frontmatter)) }
+
+  describe "Desktop navbar component" do
+    context "when rendered" do
+      it "shows background colour on link that matches the current page" do
+        desktop_navbar = page.find(".navbar__desktop")
+        home_link = desktop_navbar.find("a", text: "Home")
+        parent_li = home_link.find(:xpath, "..")
+
+        expect(parent_li["class"]).to match("active")
```

Comment:
This is fine but we can make use of a block on the expectation to test what's within, rather than starting on the inside and working outwards:

```ruby
expect(page).to have_css("li.active") do |active_li|
  expect(active_li).to have_link("Home", href: "/")
end
```